packages/react/src/components/navigationLink/NavigationLink.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './NavigationLink.module.scss';
+import classNames from '../../utils/classNames';
+import { Link } from '../link';
+
+export type NavigationLinkProps = React.PropsWithChildren<{
+  /**
+   * Link content.
+   */
+  children: string;
+  /**
+   * Additional class names to apply.
+   */
+  className?: string;
+  /**
+   * Hypertext Reference of the link.
+   */
+  href: string;
+  /**
+   * Size of the link.
+   */
+  size?: 'S' | 'M' | 'L';
```

Comment:
Should we add an as-property, for example, to replace the Link with GatsbyLink or NextLink?